src/MLOps.NET.Azure/Storage/StorageAccountMetaDataStore.cs

```diff
@@ -69,5 +73,50 @@ private async Task<CloudTable> GetTable(string tableName)
 
             return result.Result as TEntity;
         }
+
+        private async Task<IEnumerable<TEntity>> RetrieveEntities<TEntity>(string propertyName, Guid propertyValue, string tableName) where TEntity : ITableEntity, new()
+        {
+            var tableClient = storageAccount.CreateCloudTableClient(new TableClientConfiguration());
+            var items = new List<TEntity>();
+            Action<IList<TEntity>> onProgress = null;
```

Comment:
I think we should be able to simplify this not to use do-while or Actions and cancellation tokens. Happy to bounce some ideas with you if you like.

Reply:
We should be able to get the data using DbContext itself same as SQLite implementation, the only change would be to use Async methods